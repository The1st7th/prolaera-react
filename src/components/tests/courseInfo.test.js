import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../components/button';
import CourseInfo from '../../components/course/courseInfo';
import setInnerHtml from '../../helpers/domHelpers';

Enzyme.configure({ adapter: new Adapter() });

const course = {
  by: 'John Doe',
  course_id: '1234-1234-1234-1234',
  courseAudience: 'Accountants',
  delivery_method: 1,
  hours: [
    {
      credits: 1,
      subject_area: 'Tax Preparation'
    }
  ],
  level: 'Basic',
  name: 'Tax Preparation 102',
  objectives: '<ol><li>Learn about tax planning</li><li>Apply skills</li><li>Review tax planning</li></ol>',
  prep: '<p>Read course introduction</p>',
  prerequisites: '<p>Tax Prep 101</p>',
  summary: '<p>This course is the second installation of Tax Prep focused on general tax planning.</p>',
  header: 'Your course Tax Preparation 101 is ready for review',
  footer: 'Log in below',
  buttonText: 'Test Button Text',
  adminProfileId: 'testAdminProfileId',
  buttonLink: 'testButtonLink'
};

describe('CourseInfo component custom input tests', () => {
  it('creates and checks a snapshot of CourseInfo html', async () => {
    let courseHtml = renderer.create(<CourseInfo {...course} />);
    let courseJson = courseHtml.toJSON();
    expect(courseJson).toMatchSnapshot();
  });

  it('checks all custom inputs', async () => {
    const wrapper = shallow(<CourseInfo {...course} />);
    expect(wrapper.contains(<em>Tax Preparation 102 (Group-Live)</em>)).toBe(true);
    expect(
      wrapper.contains(
        <h4>
          By: <span style={{ fontWeight: 'normal' }}>John Doe</span>
        </h4>
      )
    ).toBe(true);

    expect(wrapper.contains(<li style={{ fontWeight: 'normal' }}>Tax Preparation - 1 Hour(s)</li>)).toBe(true);

    expect(
      wrapper.contains(
        <h4>
          Price: <span style={{ fontWeight: 'normal' }}>Free</span>
        </h4>
      )
    ).toBe(true);

    expect(
      wrapper.contains(
        <h4>
          Target Audience: <span style={{ fontWeight: 'normal' }}>Accountants</span>
        </h4>
      )
    ).toBe(true);
    expect(
      wrapper.contains(
        <div
          className="innerHtmlStyles"
          style={{ fontWeight: 'normal' }}
          dangerouslySetInnerHTML={setInnerHtml('<p>Read course introduction</p>')}
        />
      )
    ).toBe(true);
    expect(
      wrapper.contains(
        <div
          className="innerHtmlStyles"
          style={{ fontWeight: 'normal' }}
          dangerouslySetInnerHTML={setInnerHtml('<p>Tax Prep 101</p>')}
        />
      )
    ).toBe(true);
    expect(
      wrapper.contains(
        <h4>
          Level: <span style={{ fontWeight: 'normal' }}>Basic</span>
        </h4>
      )
    ).toBe(true);
    expect(
      wrapper.contains(
        <div
          className="innerHtmlStyles"
          style={{ fontWeight: 'normal' }}
          dangerouslySetInnerHTML={setInnerHtml(
            '<ol><li>Learn about tax planning</li><li>Apply skills</li><li>Review tax planning</li></ol>'
          )}
        />
      )
    ).toBe(true);

    expect(
      wrapper.contains(
        <div
          className="innerHtmlStyles"
          style={{ fontWeight: 'normal' }}
          dangerouslySetInnerHTML={setInnerHtml(
            '<p>This course is the second installation of Tax Prep focused on general tax planning.</p>'
          )}
        />
      )
    ).toBe(true);
    expect(
      wrapper.contains(
        <Button
          text={'Test Button Text'}
          link={'https://app.prolaera.com/#/admin/testAdminProfileId/testButtonLink/review'}
        />
      )
    ).toBe(true);
  });
});
