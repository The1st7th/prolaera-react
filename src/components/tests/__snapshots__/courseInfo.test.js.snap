// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseInfo component custom input tests creates and checks a snapshot of CourseInfo html 1`] = `
<div
  id="subHeaderWrapper"
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "display": "block",
      "fontFamily": "source-sans-pro, sans-serif",
      "fontWeight": "100",
      "marginBottom": "0px",
      "marginLeft": "auto",
      "marginRight": "auto",
      "padding": "20px",
      "textAlign": "center",
    }
  }
>
  <table
    align="center"
    border="0"
    cellPadding={0}
    cellSpacing={0}
    className="courseReview"
    valign="top"
    width="100%"
  >
    <tbody>
      <tr>
        <td
          align="left"
          style={
            Object {
              "padding": "10px 10px 0px",
            }
          }
        >
          <h3
            align="center"
            style={
              Object {
                "fontSize": "16pt",
                "marginTop": "5px",
              }
            }
          >
            Course Information:
          </h3>
          <h4
            align="center"
            style={
              Object {
                "marginBottom": "0px",
                "marginTop": "0px",
              }
            }
          >
            <a
              href="https://app.prolaera.com/#/courses/1234-1234-1234-1234"
              style={
                Object {
                  "color": "#2F4050",
                  "height": "100%",
                  "textDecoration": "underline",
                }
              }
            >
              <em>
                Tax Preparation 102
              </em>
            </a>
          </h4>
        </td>
      </tr>
      <tr>
        <td
          align="left"
        >
          <div
            style={
              Object {
                "paddingLeft": "40px",
              }
            }
          >
            <h4
              style={
                Object {
                  "marginBottom": "0px",
                }
              }
            >
              By:
               
              <span
                class="innerHtmlStyles"
                style={
                  Object {
                    "fontWeight": "normal",
                  }
                }
              >
                John Doe
              </span>
            </h4>
            <h4
              style={
                Object {
                  "marginBottom": "0px",
                }
              }
            >
              Description: 
            </h4>
            <div
              className="innerHtmlStyles"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>This course is the second installation of Tax Prep focused on general tax planning.</p>",
                }
              }
              style={
                Object {
                  "fontWeight": "normal",
                }
              }
            />
            <h4
              style={
                Object {
                  "marginBottom": "0px",
                }
              }
            >
              Learning Objectives:
            </h4>
            <div
              className="innerHtmlStyles"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<ol><li>Learn about tax planning</li><li>Apply skills</li><li>Review tax planning</li></ol>",
                }
              }
              style={
                Object {
                  "fontWeight": "normal",
                }
              }
            />
            <h4
              style={
                Object {
                  "marginBottom": "0px",
                }
              }
            >
              Delivery Method:
            </h4>
            <div
              className="innerHtmlStyles"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Group-Live",
                }
              }
              style={
                Object {
                  "fontWeight": "normal",
                }
              }
            />
            <h4
              style={
                Object {
                  "marginBottom": "0px",
                }
              }
            >
              Price:
               
              <span
                class="innerHtmlStyles"
                style={
                  Object {
                    "fontWeight": "normal",
                  }
                }
              >
                Free
              </span>
            </h4>
            <h4
              style={
                Object {
                  "marginBottom": "0px",
                }
              }
            >
              Recommended CPE Credit(s):
            </h4>
            <ul
              style={
                Object {
                  "marginTop": "10px",
                }
              }
            >
              <li
                style={
                  Object {
                    "fontWeight": "normal",
                  }
                }
              >
                Tax Preparation
                 - 
                1
                 Hour(s)
              </li>
            </ul>
            <h4
              style={
                Object {
                  "marginBottom": "0px",
                }
              }
            >
              Level:
               
              <span
                class="innerHtmlStyles"
                style={
                  Object {
                    "fontWeight": "normal",
                  }
                }
              >
                Basic
              </span>
            </h4>
            <h4
              style={
                Object {
                  "marginBottom": "0px",
                }
              }
            >
              Target Audience: 
            </h4>
            <div
              className="innerHtmlStyles"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Accountants",
                }
              }
              style={
                Object {
                  "fontWeight": "normal",
                }
              }
            />
            <h4
              style={
                Object {
                  "marginBottom": "0px",
                }
              }
            >
              Prerequisites: 
            </h4>
            <div
              className="innerHtmlStyles"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>Tax Prep 101</p>",
                }
              }
              style={
                Object {
                  "fontWeight": "normal",
                }
              }
            />
            <h4
              style={
                Object {
                  "marginBottom": "0px",
                }
              }
            >
              Prep: 
            </h4>
            <div
              className="innerHtmlStyles"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>Read course introduction</p>",
                }
              }
              style={
                Object {
                  "fontWeight": "normal",
                }
              }
            />
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            style={
              Object {
                "backgroundColor": "#72C02C",
                "borderRadius": "3px",
                "color": "#ffffff",
                "display": "inline-block",
                "margin": "auto",
                "padding": "0px",
                "textAlign": "middle",
              }
            }
          >
            <div
              align="center"
              className="customButton"
              style={
                Object {
                  "height": "100%",
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <a
                href="https://app.prolaera.com/#/admin/testAdminProfileId/testButtonLink/review"
              >
                <button
                  className="buttonWidths"
                  style={
                    Object {
                      "borderRadius": "3px",
                      "color": "#ffffff",
                      "height": "100%",
                      "padding": "14px 14px 14px 14px",
                      "textDecoration": "none",
                      "width": "100%",
                    }
                  }
                >
                  Test Button Text
                </button>
              </a>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
